cmd/vicadmin/server.go

```diff
@@ -46,10 +55,78 @@ const (
 	formatZip
 )
 
+const sessionExpiration = time.Hour * 24
+const sessionCookieKey = "sessiondata"
+
+// UserSession holds a user's session metadata
+type UserSession struct {
+	username string
+	created  time.Time
+	config   *session.Config
+}
+
+// UserSessionStore holds and manages user sessions
+type UserSessionStore struct {
+	*sync.Mutex
+	sessions map[string]*UserSession
+	ticker   *time.Ticker
+}
+
+func (u *UserSessionStore) Add(username string, config *session.Config) *UserSession {
+	u.Lock()
+	defer u.Unlock()
+	u.sessions[username] = &UserSession{
```

Comment:
Have you considered to store a UserSession into variable and then return it? Double map look ups are not good.